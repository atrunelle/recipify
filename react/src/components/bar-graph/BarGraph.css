.bar-graph {
  font-weight: 700;
  text-align: center;
  padding: 20px;
  color: #fff; }
  .bar-graph--fat {
    background: #FF5722; }
  .bar-graph--protein {
    background: #FFC107; }
  .bar-graph--carbs {
    background: #3F51B5; }
