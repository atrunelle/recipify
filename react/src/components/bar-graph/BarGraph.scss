.bar-graph {
  font-weight: 700;
  text-align: center;
  padding: 20px;
  color: #fff;
  
  &--fat {
    background: #FF5722;      
  }
  
  &--protein { 
    background: #FFC107;      
  }
  
  &--carbs {
    background: #3F51B5;      
  }
}