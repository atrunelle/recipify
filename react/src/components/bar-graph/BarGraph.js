import React, { Component } from 'react';
import './BarGraph.css';

class BarGraph extends Component {

  get classes () {
    return `bar-graph bar-graph--${this.props.label.toLowerCase()}`;
  }

  get width () {
    return `${this.props.percentage}%`;
  }

  render() {
    if (!this.props.percentage) return ('');

    return (
    <div
      style={{width : `${this.width}`}}
      className={this.classes}>
      <p>{this.props.label} {this.width}</p>
    </div>
    );
  }
}

export default BarGraph;