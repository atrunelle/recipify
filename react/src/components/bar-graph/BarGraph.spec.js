import React from 'react';
import { mount } from 'enzyme';
import BarGraph from './BarGraph';

describe('Component: BarGraph', () => {
  it('should not render if percentage is 0', () => {
    const component = mount(
      <BarGraph
        label="Fat"
      />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const component = mount(
      <BarGraph
        label="Fat"
        percentage={10}
      />);
    expect(component.html()).toMatchSnapshot();
  });
});