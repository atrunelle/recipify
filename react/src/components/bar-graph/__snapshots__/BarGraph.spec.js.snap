// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BarGraph should match snapshot 1`] = `"<div style=\\"width: 10%;\\" class=\\"bar-graph bar-graph--fat\\"><p>Fat 10%</p></div>"`;
